kolibri/plugins/learn/assets/src/learn.js

```diff
@@ -1,51 +1,19 @@
-'use strict';
-
-var _ = require('lodash');
-var logging = require('loglevel');
-
-// This is aliased for your convenience!
-var KolibriModule = require('kolibri_module');
-
-var LearnModule = KolibriModule.extend({
-
-    start: function(plugin) {
-        // This is called on the kolibri_register event - make sure it is this plugin that has registered.
-        if (this === plugin) {
-            logging.info('Learn Started');
-            var self = this;
-            document.addEventListener('DOMContentLoaded', function(event) {
-                self.render();
-            });
-        }
-    },
-    render: function() {
-        var content = document.getElementById('content-main');
-        var render_area = document.getElementById('content-render');
-        var buttons = [
-            {
-                id: 'video',
-                event: 'video/mp4'
-            },
-            {
-                id: 'audio',
-                event: 'audio/mp3'
-            },
-            {
-                id: 'pdf',
-                event: 'document/pdf'
-            }
-        ];
-        var self = this;
-        buttons.forEach(function(button) {
-            var button_node = document.createElement('button');
-            button_node.id = button.id;
-            button_node.innerHTML = button.id;
-            button_node.onclick = function () {
-                self.trigger('content_render:' + button.event, button, render_area);
-            };
-            content.appendChild(button_node);
-        });
-    }
+
+import logging from 'loglevel';
+
+
+import KolibriModule from 'kolibri_module';
+import Vue from 'vue';
+import Main from './Main.vue';
+
+
+const vm = new Vue({
```

Comment:
That seems like a reasonable way forward to me.

What I should have said was - not sure how this is plugging into the core app's Vue module (which is presumably where the Vue module should be instantiated for global control).

The reason being is that there are going to be things like, adding a plugin to the Vue global that we would want to do in the core app, and then export to all the others.

See for example, how I have exposed the Backbone library on the kolibriGlobal object (if you want to do the same kind of import substitution for Vue, that is also a way to do it - see `webpack.config.js` for details).

This would also mean that we could define the `base-template` component in the core-app against this global Vue module, I think, right?